### PR TITLE
Improve outputs of project inspection CLI commands

### PR DESCRIPTION
**Pull request by juhoinkinen**
- Makes the column widths of the output of `annif list-projects` and `annif list-vocabs` dynamic to not spend unnecessary space in terminal. For vocabularies the benefit of this is not so great, but implementing this for both listing commands  keeps the functionalities aligned. The separator for columns is 2 spaces.

- Adds vocabulary ID and modification time columns to the output of `list-projects` implementing issue #686.

- Converts the project modification timestamps to current timezone (from UTC) and formats them to YYYY-MM-DD hh:mm:ss in the outputs of `list-projects` and `show-project`, using `-` in place of `None`.

- Adds "Backend" line the output of `show-project`

An example output of `list-projects`:
```
Project ID                Project Name                   Vocabulary ID  Language  Trained  Modification time  
--------------------------------------------------------------------------------------------------------------
tfidf-fi                  TF-IDF Finnish                 yso            fi        True     2023-04-21 10:28:37
tfidf-sv                  TF-IDF Swedish                 yso            sv        True     2023-04-21 10:30:17
tfidf-en                  TF-IDF English                 yso            en        True     2023-04-21 10:30:45
fasttext-fi               fastText Finnish               yso            fi        True     2023-04-21 10:33:16
fasttext-sv               fastText Swedish               yso            sv        True     2023-04-21 10:32:25
fasttext-en               fastText English               yso            en        True     2023-04-21 10:31:28
omikuji-parabel-fi        Omikuji Parabel Finnish        yso            fi        True     2023-04-21 10:35:14
omikuji-parabel-sv        Omikuji Parabel Swedish        yso            sv        True     2023-04-21 10:32:12
omikuji-parabel-en        Omikuji Parabel English        yso            en        True     2023-04-21 10:31:47
mllm-fi                   YSO MLLM Finnish               yso            fi        False    -                  
mllm-en                   YSO MLLM English               yso            en        False    -                  
mllm-sv                   YSO MLLM Swedish               yso            sv        False    -                  
yake-fi                   YAKE Finnish                   yso            fi        True     2023-04-21 10:23:56
ensemble-fi               Ensemble Finnish               yso            fi        False    2023-04-21 10:33:16
ensemble-sv               Ensemble Swedish               yso            sv        False    2023-04-21 10:32:25
ensemble-en               Ensemble English               yso            en        False    2023-04-21 10:31:28
ykl-omikuji-parabel-fi    YKL Omikuji Parabel Finnish    ykl            fi        False    -                  
thema-omikuji-parabel-fi  Thema Omikuji Parabel Finnish  thema-fi       fi        False    -                  
```

An example output of `list-vocabs`:
```
Vocabulary ID  Languages    Size   Loaded
-----------------------------------------
yso            en,fi,se,sv  38179  True  
ykl            -            -      False 
thema-fi       -            -      False 
```

An example output of `show-project`:
```
Project ID:        fasttext-fi
Project Name:      fastText Finnish
Language:          fi
Vocabulary:        yso
Vocab language:    fi
Access:            public
Backend:           fasttext
Trained:           True
Modification time: 2023-04-21 10:33:16
```

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -82,7 +83,22 @@
...
     modification_time = re.search(r"Modification time:\s+(.+)", result.output)
-    assert modification_time.group(1) == "None"
+    assert modification_time.group(1) == "-"
+
+
+def test_show_project_modification_time(testdatadir):
+    dirpath = os.path.join(str(testdatadir), "projects", "tfidf-fi")
+    fpath = os.path.join(str(dirpath), "test_show_project_datafile")
+    os.makedirs(dirpath)
+    open(fpath, "a").close()
```
[the remark this answers is not in the document]
Comment (juhoinkinen) [reply tests/test_cli.py:93]: I think this is a false warning.

**Comment by juhoinkinen** [comment]
I started to think adding also "Backend" column to the output of `list-projects`, but maybe that is too much detail in there. But backend could be shown by `show-projects`.

**Comment by juhoinkinen** [comment]
Yet another addition would be to dump project configuration by `show-project`.

**Comment by juhoinkinen** [comment]
> Yet another addition would be to dump project configuration to output of `show-project`.

The output for project configuration could be like this (hiding the parameters that are already shown by the above lines):
```
$ annif show-project fasttext-fi 
Project ID:        fasttext-fi
Project Name:      fastText Finnish
Language:          fi
Vocabulary:        yso
Vocab language:    fi
Access:            public
Trained:           True
Modification time: 2023-04-21 10:33:16
Parameters:
  backend:         fasttext
  analyzer:        voikko(fi)
  dim:             500
  lr:              0.25
  epoch:           30
  loss:            hs
  limit:           100
  chunksize:       24
```

**Review by osma — APPROVED** [review]
LGTM

**Comment by osma** [comment]
Maybe simplest would be to merge this as it is now, then possibly work on the project configuration display in a follow-up PR?

**Pull request merged**